### PR TITLE
add support for smart_inventory

### PR DESCRIPTION
**Pull request by bell07**
Support for smart_inventory https://github.com/bell07/minetest-smart_inventory

### CODE REVIEW

File: worldedit_gui/init.lua [hunk 1]
```diff
@@ -134,8 +134,7 @@ elseif rawget(_G, "inventory_plus") then --inventory++ installed
 			inventory_plus.set_inventory_formspec(player, get_formspec(name, page))
 		end
 	end
-elseif rawget(_G, "sfinv") then --sfinv installed (part of minetest_game since 0.4.15)
-	assert(sfinv.enabled)
+elseif rawget(_G, "sfinv") and sfinv.enabled then --sfinv installed (part of minetest_game since 0.4.15)
```
Comment (sfan5) [inline worldedit_gui/init.lua]: remove this change and instead move the smart_inventory block above the sfinv one

File: worldedit_gui/init.lua [hunk 2]
```diff
@@ -171,6 +170,52 @@ elseif rawget(_G, "sfinv") then --sfinv installed (part of minetest_game since 0
...
+	function worldedit.show_page(name, page)
+		local state = smart_inventory.get_page_state("worldedit_gui", name)
+		if state then
+			state:get("code"):set_we_formspec(page)
+			state.location.rootState:show() -- update inventory page
+		end
+	end
+
+	-- smart_inventory page callback. Contains just a "custom code" element
+	function smart_worldedit_gui_callback(state)
```
Comment (sfan5) [inline worldedit_gui/init.lua]: leaks into global namespace

**Comment by sfan5** [comment]
Found one problem:
The back button in the main menu (`worldedit_gui_exit`) doesn't work.
It's fine if it just switches back to the crafting grid.

**Comment by bell07** [comment]
Done. The exit button does the same as the crafting page selection button

**Comment by sfan5** [comment]
b259906fd0241a1b2c7fafd199c87a895aa051dc

**Pull request closed without merging**